src/librustc_mir/transform/const_prop.rs

```diff
@@ -539,13 +566,39 @@ impl<'a, 'mir, 'tcx> ConstPropagator<'a, 'mir, 'tcx> {
                         *rval = Rvalue::Aggregate(
                             Box::new(AggregateKind::Tuple),
                             vec![
-                                self.operand_from_scalar(one, substs[0].expect_ty(), span),
-                                self.operand_from_scalar(two, substs[1].expect_ty(), span),
+                                self.operand_from_scalar(
+                                    one, substs[0].expect_ty(), source_info.span
+                                ),
+                                self.operand_from_scalar(
+                                    two, substs[1].expect_ty(), source_info.span
+                                ),
                             ],
                         );
                     }
                 },
                 _ => { }
+
+            },
+            interpret::Operand::Indirect(_) => {
```

Comment:
I mean technically we could just create a `ByRef` constant from the `MPlace` here, but I don't know if the rest of const prop can handle that (also may need some eager interning like what's happening now, that I would have thought to be unnecessary if we just keep reading to immediates).